src/phoebus_boundaries/phoebus_boundaries.cpp

```diff
@@ -189,6 +189,22 @@ void ReflectOuterX1(std::shared_ptr<MeshBlockData<Real>> &rc, bool coarse) {
       });
 }
 
+std::unique_ptr<parthenon::ParticleBound,
+                parthenon::DeviceDeleter<parthenon::DevMemSpace>>
+SetSwarmIX1Outflow() {
+  return DeviceAllocate<ParticleBoundIX1Outflow>();
+}
+std::unique_ptr<parthenon::ParticleBound,
+                parthenon::DeviceDeleter<parthenon::DevMemSpace>>
+SetSwarmOX1Outflow() {
+  return DeviceAllocate<ParticleBoundOX1Outflow>();
+}
+std::unique_ptr<parthenon::ParticleBound,
+                       parthenon::DeviceDeleter<parthenon::DevMemSpace>>
+SetSwarmNoWorkBC() {
```

Comment:
For MOCMC particles, the boundaries generically do things like look at the fluid state to compute thermal spectra which isn't possible with the existing parthenon boundaries. So I sidestep those boundary calls with this `NoWork` function and update the MOCMC particles at boundaries in a separate kernel in `phoebus`.

Reply:
Right now the parthenon particles boundaries are written differently from the cell-based boundaries -- the particle boundary calls are inherited classes holding `KOKKOS_INLINE_FUNCTION` functions that are called inline by the `Swarm` internally. The function signature is fixed so I don't think there's an easy way to get application-specific variables out. This is to avoid always calling like six separate kernels over all particles. Maybe there is a better way to do this but I think the issue is more specific to my `Swarm` coding than to the more general parthenon boundaries stuff.